runtime/src/secondary_index.rs

```diff
@@ -99,12 +103,125 @@ impl SecondaryIndexEntry for RwLockSecondaryIndexEntry {
     }
 }
 
+#[derive(Debug, Default)]
+struct HierarchicalOrderedMap<L1, L2>
```

Comment:
```suggestion
struct HierarchicalOrderedMap<K, V>
```
Not clear to me what L1 and L2 mean. K, V seems pretty straightforward, though.